mods/ra/maps/allies-06b/allies06b.lua

```diff
@@ -0,0 +1,275 @@
+			player.MarkFailedObjective(InfiltrateTechCenterObj)
+		end
+	end)
+end
+
+InfiltrateRef = function()
+	Trigger.OnInfiltrated(Refinery, function()
+		player.MarkCompletedObjective(InfiltrateRefObj)
+	end)
+	Trigger.OnCapture(Refinery, function()
+		player.MarkCompletedObjective(InfiltrateRefObj)
+	end)
+	Trigger.OnKilled(Refinery, function()
+		if not player.IsObjectiveCompleted(InfiltrateRefObj) then
+			player.MarkFailedObjective(InfiltrateRefObj)
+		end
+	end)
+end
+
+Tick = function()
+	if player.HasNoRequiredUnits() then
+		player.MarkFailedObjective(InfiltrateTechCenterObj)
+	end
+
+	if DestroySovietsObj and ussr.HasNoRequiredUnits() then
+		player.MarkCompletedObjective(DestroySovietsObj)
+	end
+end
+
+WorldLoaded = function()
+	player = Player.GetPlayer("Greece")
+	ussr = Player.GetPlayer("USSR")
+
+	Trigger.OnObjectiveAdded(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "New " .. string.lower(p.GetObjectiveType(id)) .. " objective")
+	end)
+
+	Trigger.OnObjectiveCompleted(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective completed")
+	end)
+
+	Trigger.OnObjectiveFailed(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective failed")
+	end)
+
+	Trigger.OnPlayerLost(player, function()
+		Media.PlaySpeechNotification(player, "MissionFailed")
+	end)
+	Trigger.OnPlayerWon(player, function()
+		Media.PlaySpeechNotification(player, "MissionAccomplished")
+	end)
+
+	InfiltrateTechCenterObj = player.AddPrimaryObjective("Infiltrate one of the Soviet tech centers with a spy.")
+	CaptureRadarDomeObj = player.AddSecondaryObjective("Capture the Radar Dome at the shore.")
+	InfiltrateRefObj = player.AddSecondaryObjective("Infiltrate the Refinery for money.")
+
+	Camera.Position = DefaultCameraPosition.CenterPosition
+
+	if Map.LobbyOption("difficulty") ~= "hard" then
+		Utils.Do(TechLabCams, function(a)
+			Actor.Create("TECH.CAM", true, { Owner = player, Location = a.Location })
+		end)
+
+		Trigger.OnEnteredProximityTrigger(SovietDefenseCam.CenterPosition, WDist.New(1024 * 7), function(a, id)
+			if a.Owner == player then
+				Trigger.RemoveProximityTrigger(id)
+				local cam1 = Actor.Create("TECH.CAM", true, { Owner = player, Location = SovietDefenseCam.Location })
+				Trigger.AfterDelay(DateTime.Seconds(15), cam1.Destroy)
+				if not DefenseFlame1.IsDead then
+					local cam2 = Actor.Create("TECH.CAM", true, { Owner = player, Location = DefenseFlame1.Location })
+					Trigger.AfterDelay(DateTime.Seconds(15), cam2.Destroy)
+				end
+				if not DefenseFlame2.IsDead then
+					local cam3 = Actor.Create("TECH.CAM", true, { Owner = player, Location = DefenseFlame2.Location })
+					Trigger.AfterDelay(DateTime.Seconds(15), cam3.Destroy)
+				end
+			end
+		end)
+
+		if Map.LobbyOption("difficulty") == "easy" then
+			Actor.Create("Camera", true, { Owner = player, Location = Weapcam.Location })
```

Comment:
if Map.LobbyOption("difficulty") == "easy" then
			Actor.Create("Camera", true, { Owner = player, Location = Weapcam.Location })

This could be removed, because it now triggers when the radar dome is captured.